### ISSUE TITLE
Screen Reader does not announce selected/ Not selected state for buttons in "Choose a Data Source" Dialog Box.:A11y_Winforms_ScreenReader

### ISSUE THREAD

**Issue by PoojaNamde** [dotnet/winforms#14457]
**"Please do not close this bug. This bug should only be closed by TCS, C+AI Accessibility or the Divisional Driver."**

### GitHub Tags:
#A11ySev2; #A11yTCS; #WCAG4.1.2; #Win32; #DesktopApp; #Win11; #WinForms .NET Runtime; #BM_WinForms.NETRuntime_Win32_June2025; #NameRoleValue; #A11yMAS; #NVDA;

### Environment Details:
Microsoft Windows 11 Enterprise (24H2) Build on 26100.4061  

### Repro Steps:
1.	Open Visual Studio 2022.
2.	TAB to Project menu dropdown and select it (Alt + P).
3.	TAB to the 'SSIS Import and Export Wizard' and select it.
4.	TAB to Next button present in 'SQL Server Import and Export Wizard' window and hit ENTER key.
5.	TAB to "Categorized" button present in 'Choose a Data Source' Screen.
6.	Observe that the screen reader does not announce the selected state of the button.

### Actual Result:
The screen reader does not announce the selected state of the button when it is selected. The state change is not programmatically conveyed to assistive technologies.

**Same issue exists for the links in the pane.** 

### Expected Result:
The screen reader should announce the button as "Selected" or "not selected," depending on the current state. This ensures that the state change is programmatically exposed and accessible to users relying on assistive technologies.

### User Impact:
Screen reader users cannot determine the current state of the “Categorized” buttons, leading to confusion and uncertainty while interacting with toggle functionality and reducing independent usability.

### Attachment:

<img width="1366" height="729" alt="Image" src="https://github.com/user-attachments/assets/d1be0f2d-a1ab-41ce-b39f-28f5a41734cd" />

Similar issue:

<img width="642" height="412" alt="Image" src="https://github.com/user-attachments/assets/e82047aa-4838-41d3-83b4-574e287d43eb" />

**Comment by SimonZhao888** [comment]
@PoojaNamde could you please provide a simple WinForms demo that reproduces this issue? I used PropertyGrid to attempt to reproduce this issue, but I was unable to obtain the result you expected.

<img width="729" height="760" alt="Image" src="https://github.com/user-attachments/assets/e5bdfcc6-66d5-4c07-a810-16a8e8db367c" />

Additionally, only controls that implement the SelectionItemPattern will announce their selected state. Listbox, Listview, and Tree view support multiple selections, Combobox and Radio button group support single selection.

**Comment by PoojaNamde** [comment]
@SimonZhao888, could you please help me with the latest winform build to reverify this issue. If issue will not reproduce, we can close this bug after reverifying at our end. Thank you

**Comment by SimonZhao888** [comment]
@PoojaNamde you can change the target framework of the project to any version of .NET 6.0 or later. The categorized buttons of PropertyGrid are now recognized as radio buttons.

**Comment by LeafShi1** [comment]
This issue is reproducible only in projects targeting **.NET 4.8** and **.NET 4.8.1**.

Users can resolve this situation by setting 
`<runtime>
	<AppContextSwitchOverrides value=" Switch.UseLegacyAccessibilityFeatures.3=true"/>
</runtime>` 
within the project's `App.config` file

Currently, the specific location where the issue originates has been identified here:
https://devdiv.visualstudio.com/DevDiv/_git/NetFx?path=/src/NDP/fx/src/WinForms/Managed/System/WinForms/PropertyGrid.cs&version=GBmain&line=2119&lineEnd=2124&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents
whether a Button's role is exposed as a standard "Button" or a "Check button" is controlled by the `AccessibilityImprovements.Level1` setting.

**Comment by KlausLoeffelmann** [comment]
If the workaround is acceptable, we can close this.

**Comment by PoojaNamde** [comment]
@LeafShi1 If issue is not reproducing in any version of .NET 6.0 or later, we can close this bug. Could you please help us with the Sample app to verify this issue so that we close this.

**Comment by LeafShi1** [comment]
A simple app: [WinFormsApp9.zip](https://github.com/user-attachments/files/26992079/WinFormsApp9.zip)
Verification Process: 

https://github.com/user-attachments/assets/1114d4c1-7dea-4eeb-b8e9-61240bffc55d

**Comment by LeafShi1** [comment]
#Closed; #A11yFixed;
